reference/forms/types/choice.rst

```diff
@@ -160,18 +222,25 @@ Field Options
 choices
 ~~~~~~~
 
-**type**: ``array`` **default**: ``array()``
+**type**: ``array`` or ``\Traversable`` **default**: ``array()``
 
 This is the most basic way to specify the choices that should be used
 by this field. The ``choices`` option is an array, where the array key
-is the item's label and the array value is the item's value::
+is the choice's label and the array value is the choice's data::
 
     $builder->add('inStock', 'choice', array(
-        'choices' => array('In Stock' => true, 'Out of Stock' => false),
+        'choices' => array(
+            'In Stock' => true,
+            'Out of Stock' => false,
+        ),
         // always include this
         'choices_as_values' => true,
     ));
 
+The component will try to cast the choices data to string to use it in view
+format, in that case ``"0"`` and ``"1"``, but you can customize it using the
+`choice_value`_ option.
```

Comment:
*will try* is ambiguous. What if it fails? *but you can customize* When is this necessary?

This paragraph may be intrepreted as:
- this component has a random behaviour (it will try but can fail)
- customization is necessary for booleans (because the *but* part seems to be a solution to the previous sentence)